transforms/pure-component.js

```diff
@@ -0,0 +1,60 @@
+const REACT_COMPONENT_CLASS_DEFINITION = {
+  superClass: {
+    type: 'MemberExpression',
+    object: {
+      name: 'React'
+    },
+    property: {
+      name: 'Component'
+    }
+  }
+};
+
+function isRenderMethod(node) {
+  return (
+    node.type == 'MethodDefinition' &&
+    node.key.type == 'Identifier' &&
+    node.key.name == 'render'
+  );
+}
+
+const THIS_PROPS = {
+  object: {
+    type: 'ThisExpression'
+  },
+  property: {
+    name: 'props'
+  }
+};
+
+function replaceThisProps(node) {
+  return j(node)
+    .find(j.MemberExpression, THIS_PROPS)
+    .replaceWith(j.identifier('props'))
+}
+
+function buildPureComponentFunction(name, body) {
+  return j.functionDeclaration(
+    j.identifier(name),
+    [j.identifier('props')],
+    body
+  );
+}
+
+module.exports = function(file, api) {
+  const j = api.jscodeshift;
+  const f = j(file.source);
```

Comment:
please copy pasta this: https://github.com/reactjs/react-codemod/blob/master/transforms/pure-render-mixin.js#L19